bitwarden_license/src/Commercial.Infrastructure.EntityFramework/SecretsManager/Repositories/SecretRepository.cs

```diff
@@ -313,6 +313,34 @@ public async Task EmptyTrash(DateTime currentDate, uint deleteAfterThisNumberOfD
         await dbContext.SaveChangesAsync();
     }
 
+    public async Task MoveSecretsAsync(IEnumerable<Core.SecretsManager.Entities.Secret> secrets, IEnumerable<Guid> projectIds)
+    {
+        using var scope = ServiceScopeFactory.CreateScope();
+        var dbContext = GetDatabaseContext(scope);
+
+        var projectSecretsSet = dbContext.Set<Dictionary<string, object>>("ProjectSecret");
+
+        using var transaction = await dbContext.Database.BeginTransactionAsync();
+
+        var secretIds = secrets.Select(s => s.Id).ToList();
+
+        var projectSecrets = secretIds.SelectMany(s => projectIds.Select(p => new Dictionary<string, object>
+        {
+            ["ProjectsId"] = p,
+            ["SecretsId"] = s
+        }));
+
+        await projectSecretsSet
+            .Where(ps => secretIds.Contains((Guid)ps["SecretsId"]))
+            .ExecuteDeleteAsync();
+
+        projectSecretsSet.AddRange(projectSecrets);
+
+        await dbContext.SaveChangesAsync();
+
+        await transaction.CommitAsync();
+    }
```

Comment:
This is cool, I didn't know you could do this.

Is the warning [here](https://learn.microsoft.com/en-us/ef/core/modeling/relationships/many-to-many#basic-many-to-many) applicable to what we're doing? 

```
Currently, EF Core uses Dictionary<string, object> to represent join entity instances for which no .NET class has been configured. 
However, to improve performance, a different type may be used in a future EF Core release. 
Do not depend on the join type being Dictionary<string, object> unless this has been explicitly configured.
```

Another option is, we could create a class for the join entity and add that to our EF config.

Reply:
I think it would have been applicable. I did decide to go the route of defining it as an entity. I did initially try that but it messed with the model a lot so I ditched it but I figured out how to do it better and there are only very minor changes to the model. Essentially now it just have a namespace name instead of an auto generated one.